src/node_crypto.cc

```diff
@@ -5725,6 +5725,17 @@ void Initialize(Local<Object> target,
 #endif  // OPENSSL_NO_SCRYPT
 }
 
+const std::string GetOpenSSLVersion() {
+  // sample openssl version string format
+  // for reference: "OpenSSL 1.1.0i  14 Aug 2018"
+  std::string ssl(OPENSSL_VERSION_TEXT);
+  size_t first = ssl.find(" ");
+  size_t second = ssl.find(" ", first + 1);
+  CHECK_GT(second, first);
+  std::string result = ssl.substr(first + 1, second - first - 1);
```

Comment:
Nit: Can we simply return the result of `substr`?

Reply:
@thefourtheye - please see 
https://github.com/nodejs/node/pull/23050#discussion_r219908302 @refack 's response to the  same: looks like that may break in c++14 world. I don't know it myself, so following SME opinion. Hope you are good with that?

Reply:
It looks like @refack is actually okay with that suggestion, can you please confirm @refack? According to https://en.wikipedia.org/wiki/Copy_elision#Summary,

> an implementation may omit a copy operation resulting from a return statement, even if the copy constructor has side effects.

So, there is no guarantee that a copy will be made and we should still be fine here.

Reply:
@gireeshpunathil @thefourtheye I think @refack’s comment was agreeing that this might otherwise not elide if we don’t return the result directly. And it’s unavoidable to create a new string object here, since `substr()` already does that.

But either way: This is not performance-critical code. Just go with the simplest form. :)

Reply:
thanks, reduced to the simplest form.

Reply:
Ohh yes please. Copy elision is a good thing that we may get for free in C++17, but for now we need to be more explicit.